device-connectors/src/testflinger_device_connectors/devices/muxpi/muxpi.py

```diff
@@ -159,8 +161,28 @@ def provision(self):
                 'the "provision_data" section of '
                 "your job_data"
             )
-        cmd = self.config.get("control_switch_local_cmd", "stm -ts")
-        self._run_control(cmd)
+        if "media" not in self.job_data["provision_data"]:
+            media = None
+            self.test_device = self.config["test_device"]
+            cmd = self.config.get("control_switch_local_cmd", "stm -ts")
+            self._run_control(cmd)
+        else:
+            media = self.job_data["provision_data"]["media"]
+            match media:
+                case "sd":
+                    self.test_device = "/dev/sd-disk"
+                case "usb":
+                    self.test_device = "/dev/tc-disk"
```

Comment:
[This is something I've recently implemented in Zapper](https://github.com/canonical/zapper/blob/7351643f097d498673375f2309be654ed325839e/zapper-control/zapper/provisioning/download.py#L61-L77), not exposed via API since it's part of the provisioning package, but it could be useful for the time being. Basically it hotplugs the device on the Zapper side while monitoring new device add events via udev.

Reply:
Thanks for this comment, I didn't know we could use something like this, but that would be definitely better! What would be the ETA to get this?

Reply:
Let's see how it goes https://github.com/canonical/zapper/pull/240

Reply:
The new API/CLI is ready and available in the latest Zapper snap.